src/nvim/main.c

```diff
@@ -250,6 +250,11 @@ int main(int argc, char **argv)
 
   argv0 = argv[0];
 
+  if (!appname_is_valid()) {
```

Comment:
no need to do this check so early. Instead do it _just_ before $NVIM_APPNAME is evaluated. That should help avoid order-of-operations issues.

Reply:
The first call to get_appname is performed in main (src/nvim/main.c:286) -> early_init (src/nvim/main.c:222) -> set_init_1 (src/nvim/option.c:340) -> stdpaths_user_state_subpath  (src/nvim/os/stdpaths.c:202) -> get_xdg_home (src/nvim/os/stdpaths.c:138.). This doesn't seem to be much later than the current call to appname_is_valid and I'm not sure getting closer to this call would change anything (but I could very well be missing something).